### DISCUSSION TITLE
Fedora: regexps lists not working

### DISCUSSION THREAD

**Discussion by tQ6yf4Mc** [evilsocket/opensnitch#1193] [General]
Hi, thank you for your great work. _I was shadow banned for a time due to my totally incognito set up account so I am bumping my issue._

So hosts format lists based blocking works without any problem (whether system-wide DNS query-based blocks or for individual apps).
_However, no one clever uses hosts format lists anymore as this format is very outdated, circa 2005 poor amateur internet cowboy, while you need to specify all sub-domains instead of using just `*`.domain.com / domain.com like in dnscrypt-proxy [it assumes ^(.`*\`.)`?`domain`\`.com$ automatically]. You can also use just `*`telemetry`*` (it is .`*`telemetry.`*`) or pixel`*` in dnscrypt-proxy. It is almost all anybody needs.
Regular expressions are of course more powerful but my blocking list for dnscrypt-proxy in Windows has >10MB and it works super fast without any problem at all. I can shut down for example 50% of Microsoft tracking, even without firewall, with just `*`microsoft`*`, `*`office`*`, `*`bing`*` and `*`azure`*` - and I have many methods if I really want to check something at microsoft.com in browsers._
So you just choose modern "wildcard domains" format per for example superb github.com/hagezi/dns-blocklists?tab=readme-ov-file#pro and that's all.

So **OpenSnitch is working great** (but you can not rearrange columns in tabs and automatically adjust / readjust columns widths. Check NetLimiter for Windows - it's very well done there. Also please add a few words to Preferences>Database options as I really did tried to understand but really don't know what some of these options do.)

**RegExps logic processing works generally here** as it works properly in [Rule>Network>To_this_host] (using full regexps notation) and also for regexps for example in [Rule>Network>Protocol] and [Rule>Network>To_this_port].
**But they do not work when list is used as in** [...>To_this_list_of_domains_(regular_expressions)]. All is ok with folder and <file.txt>. I saw in logs that OpenSnitch reports changes done in the file - how many new regexps were found. However, It does not use those regexps, neither full regexps (^(.`*\`.)`?`example`\`.com$) nor plain domains (example.com).

???

_________________________________
- Fedora Gnome newest (40, but also 39), kernel newest (was 6.8.7-300.fc40.x86_64 - even newer now).
- OpenSnitch most new rpms.
- `opensnitchd -check-requirements` OK (for 39 but I think for my fresh 40 should be too).
- Nothing interesting in 'debug' type log I think, though please write what to look for. It confirms for example that there are 5 regexps entered in the file but does not process them at all.

**Comment by tQ6yf4Mc** [comment]
Does bumping work here as my issue was invisible for a time unfortunately...

**Reply by gustavo-iniguez-goya** [reply]
Hi @tQ6yf4Mc , really sorry for not answering you before.

You didn't say what distro and opensnitch version you're using. I assume you're using v1.6.6

> But they do not work when list is used as in [...>To_this_list_of_domains_(regular_expressions)]. All is ok with folder and <file.txt>. I saw in logs that OpenSnitch reports changes done in the file - how many new regexps were found. However, It does not use those regexps, neither full regexps (^(.*\.)?example\.com$) nor plain domains (example.com).

I'd suggest to write just 1 regexp to "file.txt" (`^(.*\.)?example\.com$`), and try to progress from there. For example, let's create a priority rule that will be evaluated before any other rule:

Name: 00000-aa-test-regexp
Enable: [x]
Priority: [x]
To this regexp list: /tmp/lists


tested with v1.7.0 (latest sources as of today) and v1.6.6:

```
[2024-09-19 20:29:07.689312]  INF  [notification] change rule: [Enabled] 00000-aa-test-regexp: if(lists.domains_regexp is '/tmp/regexp'){ reject always } 1726777747688763
[2024-09-19 20:29:07.689453]  INF  loading domains lists: lists, lists.domains_regexp, /tmp/regexp
[2024-09-19 20:29:07.689535]  DBG  Operator compiled: lists.domains_regexp is '/tmp/regexp'
[2024-09-19 20:29:07.689874]  IMP  Ruleset changed due to 00000-aa-test-regexp.json, reloading ...
[2024-09-19 20:29:07.689563]  INF  monitor lists started: /tmp/regexp
[2024-09-19 20:29:07.690005]  INF  clearing domains lists: 0 - /tmp/regexp
[2024-09-19 20:29:07.720671]  DBG  Loading regexp list: /tmp/regexp/file.txt, size: 23
[2024-09-19 20:29:07.720723]  INF  1 regexps loaded, /tmp/regexp/file.txt
```

```bash
~ $ telnet www.example.com
telnet www.example.com
Trying 93.184.215.14...
Connection failed
~ $

[2024-09-19 20:29:57  DBG  Regexp list match: www.example.com, ^(.*\.)?example\.com$
[2024-09-19 20:29:57]  DBG  ✘ /usr/bin/inetutils-telnet -> 33390:fc11:4514:1919:810::1 => www.example.com (2606:2800:21f:cb07:6820:80da:af6b:8b2c):23, mark: 0 (00000-aa-test-regexp)

~ $ telnet example.com
Trying 93.184.215.14...
Connection failed
~ $

[2024-09-19 20:35:03]  DBG  Regexp list match: example.com, ^(.*\.)?example\.com$
[2024-09-19 20:35:03]  DBG  ✘ /usr/bin/inetutils-telnet -> 57724:fc11:4514:1919:810::1 => example.com (2606:2800:21f:cb07:6820:80da:af6b:8b2c):23, mark: 0 (00000-aa-test-regexp)
```

try to replicate the same example.
If it doesn't work, post the command you executed, the rule and the logs.
If it works, add more regexp to the file, or add more options to filter by.

**Comment by tQ6yf4Mc** [comment]
Hello, the issue still exists in 1.8.0 unfortunately. Fresh Fedora 43, what is the problem here ?

If I use just a rule blocking ALL (hosts and DNS resolution) like this ("To this host" option): `^((.*\.)?github\.com|.*dns.*)$` it WORKS.
If I use the rule blocking the same lists of regexps as a file ("To this list of domains" option) it does NOT work.

Please HELP, do not touch basics, it's OpenSnitch's problem of some kind. I use VPN, which is redirecting DNS to special, the same IP always.

PS. It is showing it is reloading the file in logs (when I change it) like in your comment above showing how many regexps there are in the file now, HOWEVER blocking does not work.

**Reply by gustavo-iniguez-goya** [reply]
hi @tQ6yf4Mc !

Can you share the rule and the file with the regexp? right click on the rule -> export -> to clipboard.

Perhaps we're overlooking something.
